src/openrct2/peep/Peep.cpp

```diff
@@ -2862,8 +2862,9 @@ static void peep_interact_with_path(Peep* peep, const CoordsXYE& coords)
  *
  *  rct2: 0x00693F70
  */
-static bool peep_interact_with_shop(Peep* peep, int16_t x, int16_t y, TileElement* tile_element)
+static bool peep_interact_with_shop(Peep* peep, const CoordsXYE& coords)
 {
+    auto tile_element = coords.element;
     ride_id_t rideIndex = tile_element->AsTrack()->GetRideIndex();
```

Comment:
This is the only place `tile_element` is used. Rather than creating an intermediate variable, just do 
```suggestion
    ride_id_t rideIndex = coords.element->AsTrack()->GetRideIndex();
```